ssl/statem/statem_clnt.c

```diff
@@ -1122,6 +1122,7 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
          * overwritten if the server refuses resumption.
          */
         if (s->session->session_id_length > 0) {
+            s->ctx->stats.sess_miss++;
```

Comment:
While I'm not super-familiar with this code, ssl_sess.c is already accessing session_id_length and sess_miss with no immediately visible locking.  (Presumably the larger-scope locks like SSL_CTX are not going to be in use here...)